### PR TITLE
rgw: fix rgw crash when client post object with null conditions

### PR DESCRIPTION
**Pull request by diluga**
Fixes: http://tracker.ceph.com/issues/17635

Signed-off-by: Feng Guo <diluga@gmail.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
@diluga next time, you can `git push -f` to your remote branch to update the pull request. no need to post a new PR to address the reviewer's comment. =)

File: src/rgw/rgw_policy_s3.cc [hunk 1]
```diff
@@ -276,6 +276,10 @@ int RGWPolicy::from_json(bufferlist& bl, string& err_msg)
       int i;
       for (i = 0; !citer.end() && i < 3; ++citer, ++i) {
 	JSONObj *o = *citer;
+        const string condition_data = o->get_data();
```
Comment (yehudasa) [inline src/rgw/rgw_policy_s3.cc]: @diluga make it `const string&`

**Comment by yehudasa** [comment]
@diluga does that fix it for you? Please see my fix here: https://github.com/yehudasa/ceph/tree/wip-11573-hammer

**Comment by yehudasa** [comment]
@diluga should be this: https://github.com/yehudasa/ceph/tree/wip-17635-hammer

File: src/rgw/rgw_policy_s3.cc (hunk 1 above)
Comment (diluga) [inline src/rgw/rgw_policy_s3.cc]: thx！i have chage the code，push  it again！

**Comment by diluga** [comment]
@yehudasa https://github.com/yehudasa/ceph/tree/wip-17635-hammer i have test it，it worked！

**Comment by oritwas** [comment]
fixed in https://github.com/ceph/ceph/pull/11581

**Pull request closed without merging**